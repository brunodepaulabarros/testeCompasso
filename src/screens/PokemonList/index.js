import React, { useEffect, useState } from 'react';
import { ActivityIndicator, FlatList, Image, Text, View } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import LoadCard from '../../components/LoadCard';
import PokemonBaner from '../../components/PokemonBaner';
import { pokemonRequest } from "../../services/redux/actions/pokemonRequestAction";
import { getListOfPokemons, getPokemonById } from '../../services/requests/PokemonRequest';
import {
    Container,
    ContainerHeader
} from './styles';

const PokemonList = () => {

    const [pokemonList, setPokemonList] = useState([]);
    const [loadingPagination, setLoadingPagination] = useState(true);

    const pokemonRequestUrl = useSelector(state => state.pokemonRequestReducer)
    const dispatch = useDispatch();

    const addUrlNext = (data) => {
        dispatch(pokemonRequest(data))
    }

    useEffect(() => {
        loadApp()
    }, [])

    const loadApp = async () => {
        response = await getListOfPokemons()
        var pokeList = [];
        for (var i = 0; i < response.data.results.length; i++) {
            var _response = await getPokemonById(response.data.results[i].url)
            pokeList.push(_response.data)
        }
        addUrlNext(response.data)
        setPokemonList(pokeList)
    }

    const loadNewPage = async () => {
        setLoadingPagination(true)
        response = await getListOfPokemons(pokemonRequestUrl.urlNext)
        var pokeList = [];
        for (var i = 0; i < response.data.results.length; i++) {
            var _response = await getPokemonById(response.data.results[i].url)
            pokeList.push(_response.data)
        }
        setPokemonList(pokemonList.concat(pokeList))
        addUrlNext(response.data)
        setLoadingPagination(false)
    }

    const renderFooter = () => {
        return (
            <View >
                {loadingPagination && <ActivityIndicator size="large" color="green" />}
            </View>
        );
    };

    return (
        <Container >
            <ContainerHeader >
                <Text style={{ fontSize: 26, fontWeight: "700" }}>Pokedex</Text>
                <Image
                    style={{
                        height: 180,
                        width: 180, position: "absolute",
                        right: -40, top: -40,
                    }}
                    source={require('../../assets/images/pokeball.png')}
                />
            </ContainerHeader>
            {pokemonList.length === 0 ?
                <View>
                    <LoadCard />
                    <LoadCard />
                    <LoadCard />
                    <LoadCard />
                </View>
                :
                <View style={{ backgroundColor: "#fff", flex: 1 }}>
                    <FlatList
                        data={pokemonList}
                        keyExtractor={item => item.name}
                        showsVerticalScrollIndicator={false}
                        onEndReached={() => loadNewPage()}
                        onEndReachedThreshold={0.1}
                        style={{ flex: 1 }}
                        ListFooterComponent={() => renderFooter()}
                        renderItem={({ item }) =>
                            <PokemonBaner pokemonInfo={item} />
                        }
                    />
                </View>
            }
        </Container >
    )
}

export default PokemonList;