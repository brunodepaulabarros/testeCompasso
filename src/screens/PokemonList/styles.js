import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
    flex: 1;
    background-color: #fff; 
`;

export const ContainerHeader = styled.View`
    flex-Direction: row;
    justify-Content: space-between;
    height: 80px;
    align-Items: flex-end;
    margin:0 0 0 15px;
`;
