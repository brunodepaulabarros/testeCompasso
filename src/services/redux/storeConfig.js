import { combineReducers, createStore } from 'redux';
import pokemonRequestReducer from "./reducers/pokemonRequestReducer";

const reducers = combineReducers({
    pokemonRequestReducer,
})

const storeConfig = () => {
    return createStore(reducers)
}

export default storeConfig;