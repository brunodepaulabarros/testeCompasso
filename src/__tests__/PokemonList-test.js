import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import PokemonList from '../screens/PokemonList';
import storeConfig from '../services/redux/storeConfig';


const Store = storeConfig();
test('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={Store}>
      <PokemonList />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});