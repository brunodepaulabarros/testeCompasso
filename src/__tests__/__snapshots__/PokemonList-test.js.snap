// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
          "marginBottom": 0,
          "marginLeft": 15,
          "marginRight": 0,
          "marginTop": 0,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 26,
          "fontWeight": "700",
        }
      }
    >
      Pokedex
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/pokeball.png",
        }
      }
      style={
        Object {
          "height": 180,
          "position": "absolute",
          "right": -40,
          "top": -40,
          "width": 180,
        }
      }
    />
  </View>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#ddd",
            "borderBottomWidth": 0,
            "borderColor": "#ddd",
            "borderRadius": 15,
            "borderWidth": 2,
            "height": 150,
            "marginBottom": 15,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 30,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "marginTop": 5,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#ddd",
            "borderBottomWidth": 0,
            "borderColor": "#ddd",
            "borderRadius": 15,
            "borderWidth": 2,
            "height": 150,
            "marginBottom": 15,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 30,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "marginTop": 5,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#ddd",
            "borderBottomWidth": 0,
            "borderColor": "#ddd",
            "borderRadius": 15,
            "borderWidth": 2,
            "height": 150,
            "marginBottom": 15,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 30,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "marginTop": 5,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#ddd",
            "borderBottomWidth": 0,
            "borderColor": "#ddd",
            "borderRadius": 15,
            "borderWidth": 2,
            "height": 150,
            "marginBottom": 15,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 0,
            "paddingBottom": 20,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "margin": 5,
              }
            }
          >
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 4,
                      "height": 20,
                      "width": 120,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingHorizontal": 30,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 20,
                        "marginTop": 5,
                        "width": 80,
                      },
                      Object {
                        "backgroundColor": "#E1E9EE",
                        "overflow": "hidden",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "height": 80,
                      "width": 80,
                    },
                    Object {
                      "backgroundColor": "#E1E9EE",
                      "overflow": "hidden",
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTSafeAreaView>
`;
